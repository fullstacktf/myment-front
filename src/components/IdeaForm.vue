<template>
  <div class="formWrapper">
    <b-field horizontal label="Enter some tags">
      <b-taginput v-model="tags" :data="filteredTags"></b-taginput>
    </b-field>

    <b-field horizontal label="Idea Name" message="Here goes the name of your idea">
      <b-input name="subject" expanded></b-input>
    </b-field>

    <b-field horizontal label="Description">
      <b-input type="textarea"></b-input>
    </b-field>
    <div class="columns">
      <div class="column is-2"></div>
      <div class="column is-4">
        <b-field horizontal label="Start Time">
          <b-clockpicker v-model="minTime" :hour-format="format"></b-clockpicker>
        </b-field>
      </div>
      <div class="column is-4">
        <b-field horizontal label="End Time">
          <b-clockpicker v-model="maxTime" :hour-format="format"></b-clockpicker>
        </b-field>
      </div>
    </div>

    <b-field horizontal label="url" message="Maybe your idea is linked to a website?">
      <b-input name="subject" expanded></b-input>
    </b-field>

    <b-field horizontal class="is-fullwidth" label="Idea type">
      <b-select placeholder="Select between the types">
        <option value="1">Leisure</option>
        <option value="2">Food</option>
        <option value="3">Lodging</option>
      </b-select>
    </b-field>

    <b-field horizontal>
      <!-- Label left empty for spacing -->
      <p class="control">
        <button class="button is-primary is-fullwidth">share your idea</button>
      </p>
    </b-field>
  </div>
</template>

<script>
export default {
  name: 'IdeaForm',
  data() {
    return {
      isSelectOnly: false,
      tags: ['city', 'zone', 'street', 'number'],
      allowNew: true,
      zoom: 7,
    };
  },
  mounted() {},
};
</script>

<style>
.formWrapper {
  padding: 90px;
  padding-left: 30px;
  background-color: #a5bec4;
  min-height: 30%;
  align-self: center;
  justify-self: center;
  border-radius: 10px;
}
</style>
