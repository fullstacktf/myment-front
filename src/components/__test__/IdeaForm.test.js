import IdeaForm from '../IdeaForm.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Buefy from 'buefy';
import mock from '../../test/mockBack';

describe('IdeaForm', () => {
  let localVue;
  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    localVue = Vue;
  });
  test('render component', () => {
    const wrapper = shallowMount(IdeaForm, {
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  }),
    test('Check props', () => {}),
    test('check Contents props', () => {});
});
