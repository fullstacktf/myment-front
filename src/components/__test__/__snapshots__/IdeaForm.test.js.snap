// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaForm render component 1`] = `"<div></div>"`;
